automotive/automotive-lib/src/main/java/com/example/android/uamp/automotive/lib/ListPreference.kt

```diff
@@ -0,0 +1,320 @@
+/*
+ * Copyright 2020 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.android.uamp.automotive.lib
+
+import android.content.Context
+import android.content.res.TypedArray
+import android.os.Parcel
+import android.os.Parcelable
+import android.text.TextUtils
+import android.util.AttributeSet
+import android.util.Log
+import androidx.annotation.ArrayRes
+import androidx.core.content.res.TypedArrayUtils
+import androidx.preference.Preference
+
+/**
+ * A {@link Preference} that displays a list of entries in a {@link ListPreferenceFragment}.
+ *
+ * This class is taken largely as is from {@link androidx.preference.ListPreference} other than
+ * the modification to display entries in a full-screen fragment rather than a dialog.
+ *
+ * <p>This preference saves a string value. This string will be the value from the
+ * {@link #setEntryValues(CharSequence[])} array.
+ *
+ * @attr name android:entries
+ * @attr name android:entryValues
+ */
+class ListPreference : Preference {
+
+  companion object {
+    internal const val ARG_KEY = "AutomotiveListPreferenceKey"
+  }
+
+  private val TAG = "ListPreference"
+  private var mEntries: Array<CharSequence>? = null
+  private var mEntryValues: Array<CharSequence>? = null
+  private var mValue: String? = null
+  private var mSummary: String? = null
+  private var mValueSet: Boolean = false
+
+  constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int, defStyleRes: Int)
+    : super(context, attrs, defStyleAttr, defStyleRes) {
+    init(context, attrs, defStyleAttr, defStyleRes)
+  }
+
+  constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int)
+    : this(context, attrs, defStyleAttr, R.style.AutomotivePreference_Preference)
+
+  constructor(context: Context, attrs: AttributeSet?) : this(context, attrs, R.attr.preferenceStyle)
+
+  constructor(context: Context) : this(context, null)
+
+  private fun init(context: Context, attrs: AttributeSet?, defStyleAttr: Int, defStyleRes: Int) {
+
+    val a = context.obtainStyledAttributes(
```

Comment:
Since this is a sample, and developers may not be native speakers, it's best to use more descriptive names for variables, and avoid using acronyms except in cases that are very common/well known.

Reply:
Done